.github/api_best_practices.md

```diff
@@ -0,0 +1,231 @@
+  oneof resource {
+    string deployment_id;
+    string cluster_id;
+  }
+}
+
+```
+
+</td>
+</tr>
+</table>
+
+### gRPC Message Name
+The request and response messages **must** be named after method names with suffix `Request` and `Response` unless 
+the request/response type is an empty message. Generally, resource type as response message **should** be avoided 
+e.g. use `GetDeploymentResponse` response instead of `Deployment`. This allows augmenting the response with 
+supplemental information in the future.
+
+| Verb                      | Noun           | Method name         | Request message            | Response message            |
+|---------------------------|----------------|---------------------|----------------------------|-----------------------------|
+| List                      | Deployment     | `ListDeployments`   | `ListDeploymentRequest`    | `ListDeploymentResponse`    |
+| Get                       | Deployment     | `GetDeployment`     | `GetDeploymentRequest`     | `GetDeploymentResponse`     |   
+| Update                    | Deployment     | `UpdateDeployment`  | `UpdateDeploymentRequest`  | `UpdateDeploymentResponse`  |  
+| Delete                    | Deployment     | `DeleteDeployment`  | `DeleteDeploymentRequest`  | `google.protobuf.Empty`     |     
+| Get                       | ReportStatus   | `GetReportStatus`   | `GetReportStatusRequest`   | `GetReportStatusResponse`   |  
+| Run                       | ComplianceScan | `RunComplianceScan` | `RunComplianceScanRequest` | `RunComplianceScanResponse` | 
+
+Avoid prepositions as much as possible (e.g. “for”, “with”; `DeploymentWithProcessInfo`, `DeploymentWithImageScan`).
+In case such a need arises, add a field to the request message and response message.
+
+<table>
+<tr>
+<td><b>Instead of</b></td><td><b>Use</b></td>
+</tr>
+<tr>
+<td>
+
+`GetDeploymentWithImageScanRequest`
+
+</td>
+<td>
+
+```
+GetDeploymentRequest {
+  bool with_image_scan;
+}
+```
+
+</td>
+</tr>
+<tr>
+<td>
+
+`GetDeploymentWithImageScanResponse`
+
+</td>
+<td>
+
+```
+GetDeploymentResponse {
+  Deployment deployment;
+  Image image;
+}
+
+Or,
+
+GetDeploymentImageScanResponse {
+  Image image;
+}
+```
+
+</td>
+</tr>
+<tr>
+<td>
+
+`RunPolicyScanForDeploymentRequest`
+
+</td>
+<td>
+
+`RunDeploymentPolicyScanRequest`
+
+</td>
+</tr>
+</table>
+
+All fields in the message **must** be lowercase and underscore separated names. The JSON names for the fields are 
+autogenerated by the proto compiler. By default, field names are converted to camel case notation.
+
+| Proto field name           | JSON field name        |
+|----------------------------|------------------------|
+| `network_data_start_time`  | `networkDataStartTime` |
+| `expiry_date`              | `expiryDate`           |
+
+Be explicit about conveying the specific purpose of fields e.g. instead of `expires_on`
+use `expiry_date`(/`timestamp`), instead of `network_data_since` use `network_data_start_time`.
```

Comment:
What is the motivation for this one? I actually think that the suggested names are longer and not really more informative and hence inferior.

Reply:
Clarity is preferred. For example, `network_data_since` does not convey the unit; it may or may not include time portion.